src/coreclr/pal/src/include/pal/context.h

```diff
@@ -28,17 +28,18 @@ extern "C"
 #include <signal.h>
 #include <pthread.h>
 
-#if !HAVE_MACH_EXCEPTIONS
 /* A type to wrap the native context type, which is ucontext_t on some
  * platforms and another type elsewhere. */
 #if HAVE_UCONTEXT_T
-#include <ucontext.h>
+#include <sys/ucontext.h>
```

Comment:
@janvorli, was this part of the change required for activation via signal?

As far as I can tell, all systems that have `sys/ucontext.h` also have toolchain-specific `ucontext.h` which includes `sys/ucontext.h`, plus some additional defines. This is the case on macOS, Linux and illumos. However, on FreeBSD, it is `sys/ucontext.h` which includes `machine/ucontext.h` but it was working fine there before with `<ucontext.h>`.

I am asking as this is breaking illumos build since Friday. From [logs](https://github.com/am11/runtime/runs/1670471116?check_suite_focus=true):
```
2021-01-08T19:02:42.0116455Z   [ 40%] Building CXX object pal/src/CMakeFiles/coreclrpal.dir/thread/context.cpp.o
2021-01-08T19:02:42.2653807Z   In file included from /runtime/src/coreclr/pal/src/thread/context.cpp:25:
2021-01-08T19:02:42.2655771Z   /runtime/src/coreclr/pal/src/thread/context.cpp: In function 'void CONTEXTToNativeContext(const CONTEXT*, native_context_t*)':
2021-01-08T19:02:42.2657287Z   /runtime/src/coreclr/pal/src/include/pal/context.h:120:41: error: 'REG_RBP' was not declared in this scope
2021-01-08T19:02:42.2658120Z    #define MCREG_Rbp(mc)       ((mc).gregs[REG_RBP])
2021-01-08T19:02:42.2658595Z                                            ^~~~~~~
```

this is because register definitions are included separately after the inclusion of `sys/ucontext.h`: https://github.com/illumos/illumos-gate/blob/260693/usr/src/head/ucontext.h#L35-L48.

Reply:
Opened #46790 with a potential fix.

Reply:
Without this change, it is not compiling on macOS. I am getting:
```
In file included from /Users/janvorli/git/runtime2/src/coreclr/pal/src/exception/signal.cpp:51:
In file included from /Users/janvorli/git/runtime2/src/coreclr/pal/src/include/pal/context.h:34:
/Users/janvorli/Downloads/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk/usr/include/ucontext.h:51:2: error: The
      deprecated ucontext routines require _XOPEN_SOURCE to be defined
#error The deprecated ucontext routines require _XOPEN_SOURCE to be defined
 ^
1 error generated.

```

Reply:
On macos, the ucontext.h contains prototypes for makecontext, swapcontext etc that are deprecated. The sys/ucontext.h contains definition of the actual data structure.
Maybe it would be better to go back to including the <ucontext.h> and just define the _XOPEN_SOURCE symbol before including it.

Reply:
It seems to be compiling with SDK 10.4 locally and CI is green in the PR. Let me try with SDK 11.0.